pkg/genericapiserver/server/storage_factory.go

```diff
@@ -30,6 +30,15 @@ import (
 	"github.com/golang/glog"
 )
 
+// Backend describes the storage servers, the information here should be enough
+// for health validations
+type Backend struct {
+	Server string
+	// TLS credentials
```

Comment:
document: these will be empty if it is not an HTTPS connection.

Better yet, make a `type ClientCert {Key, Cert string}` and put here a `ClientCert *ClientCert`, so it is clear it can be nil?